operator/pkg/parsed/template_to_module.go

```diff
@@ -74,12 +74,22 @@ func TemplatesToModules(
 	return modules, nil
 }
 
+func CreateModuleName(moduleName string, kymaName string) string {
+	return moduleName + kymaName
+}
+
 func NewModule(
 	template *v1alpha1.ModuleTemplate,
 	verification signature.Verification,
 ) (*unstructured.Unstructured, error) {
-	component := &template.Spec.Data
-
+	component := template.Spec.Data.DeepCopy()
```

Comment:
@jakobmoellersap , I read your PR, here just directly return for control plane module, but I think eventually to treat them (manifest/module CR) identically should no harm, we can discuss on this.

Reply:
Agreed. I think this is what we should aim for, NewModule should just return the unstructured no matter the type